### PR TITLE
feat: Add remaining arithmetic operations

### PR DESCRIPTION
**Pull request by lovenoble**
https://github.com/Inco-fhevm/inco-monorepo/issues/6

### CODE REVIEW

File: fhevm/tee_arthmetic_gas.go [hunk 1]
```diff
@@ -39,3 +39,41 @@ func teeMulRequiredGas(environment EVMEnvironment, input []byte) uint64 {
...
+	if err != nil {
+		logger.Error("teeDiv RequiredGas() ciphertext inputs not verified", "err", err, "input", hex.EncodeToString(input))
+		return 0
+	}
+	if lhs.fheUintType() != rhs.fheUintType() {
+		logger.Error("teeDiv RequiredGas() operand type mismatch", "lhs", lhs.fheUintType(), "rhs", rhs.fheUintType())
+		return 0
+	}
+
+	return environment.FhevmParams().GasCosts.TeeDiv[lhs.fheUintType()]
```
Comment (amaury1093) [inline fhevm/tee_arthmetic_gas.go]: It seems like these codes are also repeated. Any ways to refactor them?

File: fhevm/tee_arithmetic.go [hunk 2]
```diff
@@ -110,6 +110,18 @@ func teeMulRun(environment EVMEnvironment, caller common.Address, addr common.Ad
 	})
 }
 
+func teeDivRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doArithmeticOperation("teeMulRun", environment, caller, input, runSpan, func(a uint64, b uint64) uint64 {
+		return (a - a % b) / b
```
Comment (amaury1093) [inline fhevm/tee_arithmetic.go]: Why did you choose this instead of simply a/b?

Comment (lovenoble) [reply fhevm/tee_arithmetic.go]: Ah. I mistake someting.

File: fhevm/tee_arthmetic_gas.go (hunk 1 above)
Comment (lovenoble) [reply fhevm/tee_arthmetic_gas.go]: Okay. I'll try.

File: fhevm/tee_arithmetic_gas.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package fhevm
+
+import "encoding/hex"
+
+func teeArithmeticGas(op string, environment EVMEnvironment, input []byte) uint64 {
```
Comment (amaury1093) [inline fhevm/tee_arithmetic_gas.go]: The `switch` case is a bit ugly in this function, let's change the function argument to:

```suggestion
func teeArithmeticGas(op string, environment EVMEnvironment, input []byte, gasCosts map[tfhe.FheUintType]uint64) uint64 {
```

File: fhevm/tee_arithmetic_gas.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		cost = environment.FhevmParams().GasCosts.TeeRem[lhs.fheUintType()]
+	default:
+		cost = 0
+	}
+
+	return cost
+}
+
+func teeAddSubRequiredGas(environment EVMEnvironment, input []byte) uint64 {
+	return teeArithmeticGas("teeAddSub", environment, input)
```
Comment (amaury1093) [inline fhevm/tee_arithmetic_gas.go]: Using the above suggestion, we can do:

```suggestion
	return teeArithmeticGas("teeAddSub", environment, input, environment.FhevmParams().GasCosts.TeeAddSub)
```

**Review by amaury1093 — APPROVED** [review]

**Pull request merged**